sched/distributor/poolDistributor_test.go

```diff
@@ -1,230 +1,127 @@
-package distributor
+package distributor_test
 
 import (
-	"github.com/golang/mock/gomock"
-	cm "github.com/scootdev/scoot/sched/clustermembership"
-	"strings"
+	"github.com/scootdev/scoot/cloud/cluster"
+	"github.com/scootdev/scoot/cloud/cluster/memory"
+	"github.com/scootdev/scoot/sched/distributor"
 	"testing"
 )
 
-func TestPoolDistributor_NilCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nil)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil != dist {
-		t.Error("Expected dist to be nil when no members in Cluster")
-	}
-}
-
-func TestPoolDistributor_EmptyCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	nodes := make([]cm.Node, 0)
-
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nodes)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil != dist {
-		t.Error("Expected dist to be nil when no members in Cluster")
-	}
-}
-
 func TestPoolDistributor_OneNodeCluster(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	nodes := cm.GenerateTestNodes(1)
-	clusterMock := cm.NewMockCluster(mockCtrl)
-	clusterMock.EXPECT().Members().Return(nodes)
-
-	dist := NewPoolDistributor(clusterMock)
-
-	if nil == dist {
-		t.Error("Expected dist to not be nil when 1 or more nodes in Cluster")
-	}
-
-	node1 := dist.ReserveNode()
-
-	select {
-	case <-dist.freeCh:
-		t.Error("Expected cluster to be full schedlued")
-	default:
-	}
-
-	dist.ReleaseNode(node1)
-	node2 := dist.ReserveNode()
-
-	if node1.Id() != node2.Id() {
-		t.Error("Expected nodes to be the same Id since there is only one in the cluster")
-	}
-
-	select {
-	case <-dist.freeCh:
-		t.Error("Expected cluster to be full schedlued")
-	default:
-	}
+	h := makeHelper(t, "node1")
```

Comment:
I disagree with this assessment of Mock based tests.  Since in Golang you can only Mock Interfaces, a mock is just as good as a fake, without having to support as much code.  If you refactor or change the interface of course you have to change the test.  With mocks then you just need to generate the new mock class instead of updating the fake.  

The case you cited can't happen in Golang because PoolDistributor isn't an interface.

I also find handling error cases without mocks much harder, and [testing error handling code has been shown to prevent numerous catastrophic failures in distributed systems](https://www.usenix.org/system/files/conference/osdi14/osdi14-paper-yuan.pdf).  

Mocks are pretty standard way of testing in Industry and at Twitter. I really don't understand the desire to remove all of them from the codebase.  I think we should have a meeting to discuss this because it keeps coming up in code review.